registration/models.py

```diff
@@ -398,12 +400,17 @@ class BetaTestApplication(ValidateModelMixin, TimeStampedModel):
         choices=STATUS_CHOICES,
         default=PENDING,
     )
-    instance = models.ForeignKey(
-        OpenEdXInstance,
-        on_delete=models.SET_NULL,
-        null=True,
-        blank=True,
-    )
+    # Using GenericForeignKey to support both GroveInstance as well as OpenEdxInstance types
+    instance_type = models.ForeignKey(ContentType, on_delete=models.CASCADE, null=True, blank=True)
+    instance_id = models.PositiveIntegerField(null=True, blank=True)
+    instance = fields.GenericForeignKey('instance_type', 'instance_id')
+    # Above three lines would eventually be replaced by only GroveInstance:
+    #   instance = models.ForeignKey(
+    #       GroveInstance,
+    #       on_delete=models.SET_NULL,
+    #       null=True,
+    #       blank=True,
+    #   )
```

Comment:
I thought it would be simple when we have to switch to Grove completely. But you're right, I can only keep the first line (i.e. `Above three lines would eventually be replaced by only GroveInstance`)